### PR TITLE
fix issues-2016

### PR DESCRIPTION
**Pull request by niuxiaojie81**

### CODE REVIEW

**Comment by niuxiaojie81** [comment]
[fix issues-2016](https://github.com/PlatONnetwork/PlatON-Go/issues/2016)

File: eth/downloader/downloader.go [hunk 1]
```diff
@@ -852,17 +852,23 @@ func (d *Downloader) spawnSync(fetchers []func() error) error {
 		}
 	}
 	mode := d.getMode()
+	current := d.blockchain.CurrentBlock().NumberU64()
 	if mode == FastSync {
-		if failed {
-			if err := d.setFastSyncStatus(FastSyncFail); err != nil {
-				return err
+		if failed && current <= 0 {
```
Comment (benbaley) [inline eth/downloader/downloader.go]: 这里current是uint64，应该改为 failed && current == 0

File: eth/downloader/downloader.go [hunk 2]
```diff
@@ -852,17 +852,23 @@ func (d *Downloader) spawnSync(fetchers []func() error) error {
...
+			if error := d.setFastSyncStatus(FastSyncFail); error != nil {
+				err = error
```
Comment (benbaley) [inline eth/downloader/downloader.go]:
```
if mode == FastSync {
    if failed && current == 0 {
        err = d.setFastSyncStatus(FastSyncFail)
    } else {
        err = d.setFastSyncStatus(FastSyncDel)
    }
}
if mode == FullSync {
    err = d.setFastSyncStatus(FastSyncDel)
}
```
看看这样是否可以？

Comment (niuxiaojie81) [reply eth/downloader/downloader.go]: LGTM

**Review by cheng762 — APPROVED** [review]

**Comment by benbaley** [comment]
fix conflicts in #1092

**Pull request closed without merging**